src/rfkill/rfkill.c

```diff
@@ -335,9 +335,9 @@ static int run(int argc, char *argv[]) {
                         break;
                 }
 
-                if (l < RFKILL_EVENT_SIZE_V1)
-                        return log_error_errno(SYNTHETIC_ERRNO(EIO), "Short read of struct rfkill_event: (%zd < %d)",
-                                               l, RFKILL_EVENT_SIZE_V1);
+                if ((size_t)l < RFKILL_EVENT_SIZE_V1) /* l cannot be < 0 here. Cast to fix -Werror=sign-compare */
+                        return log_error_errno(SYNTHETIC_ERRNO(EIO), "Short read of struct rfkill_event: (%zd < %lu)",
+                                               l, (unsigned long) RFKILL_EVENT_SIZE_V1); /* Casting necessary to make different archs happy */
```

Comment:
Why the format change?. RFKILL_EVENT_SIZE_V1 is defined as '8', so %d is appropriate.

Reply:
not on aarch64 - you get a warning there

Reply:
```
../src/rfkill/rfkill.c: In function â€˜runâ€™:
../src/rfkill/rfkill.c:339:70: error: format â€˜%uâ€™ expects argument of type â€˜unsigned intâ€™, but argument 8 has type â€˜long unsigned intâ€™ [-Werror=format=]
  339 |                         return log_error_errno(SYNTHETIC_ERRNO(EIO), "Short read of struct rfkill_event: (%zd < %u)",
      |                                                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/basic/log.h:195:86: note: in definition of macro â€˜log_full_errnoâ€™
  195 |                         ? log_internal(_level, _e, PROJECT_FILE, __LINE__, __func__, __VA_ARGS__) \
      |                                                                                      ^~~~~~~~~~~
../src/rfkill/rfkill.c:339:32: note: in expansion of macro â€˜log_error_errnoâ€™
  339 |                         return log_error_errno(SYNTHETIC_ERRNO(EIO), "Short read of struct rfkill_event: (%zd < %u)",
      |                                ^~~~~~~~~~~~~~~
../src/rfkill/rfkill.c:339:114: note: format string is defined here
  339 |                         return log_error_errno(SYNTHETIC_ERRNO(EIO), "Short read of struct rfkill_event: (%zd < %u)",
      |                                                                                                                 ~^
      |                                                                                                                  |
      |                                                                                                                  unsigned int
      |                                                                                                                 %lu
```

https://download.copr.fedorainfracloud.org/results/packit/systemd-systemd-19302/fedora-rawhide-aarch64/02130652-systemd/builder-live.log.gz

Reply:
Oh, the kernel changed the type: 71826654ce40112f0651b6f4e94c422354f4adb6.
The define was `int`, now it's `size_t`.
So I think we need a cast, but please do it properly, i.e. use `%zu` and `size_t`.

Reply:
done